site/public/js/user-profile.js

```diff
@@ -121,6 +128,41 @@ function updateUserProfilePhoto () {
   return false;
 }
 
+function updateUserSecondaryEmail () {
+    const second_email = $('#user-secondary-email-change');
+    const second_email_notify = $('#user-secondary-email-notify-change');
+    if (second_email.data('current-second-email') === second_email.val() && second_email_notify.get(0).checked === (second_email_notify.data('current-second-email-notify') === 1)) {
+        displayErrorMessage('No changes detected to secondary email');
+    }
+    else {
+        let data = new FormData();
+        data.append('csrf_token', $("#user-secondary-email-csrf").val());
+        data.append('secondary_email', second_email.val());
+        data.append('secondary_email_notify', second_email_notify.get(0).checked);
+        let url = buildUrl(['user_profile', 'change_secondary_email']);
+        $.ajax({
+            url,
+            type: "POST",
+            data,
+            processData: false,
+            contentType: false,
```

Comment:
Alternatively, you can set the form action to be the update secondary email endpoint and have the controller redirect the user to the current page since I don't think avoiding a refresh is super critical in this operation